src/main/antlr4/imports/mysql_idents.g4

```diff
@@ -56,4 +59,4 @@ fragment BACKTICK: '`';
 QUOTED_IDENT: BACKTICK (('``') | ~('`'))+ BACKTICK;
 
 fragment DIGIT: [0-9];
-WS  :   [ \t\n\r]+ -> skip ;
+WS  :   [ \t\n\r]+ -> channel(HIDDEN) ;
```

Comment:
so when tokens (like our WS token here) are recognized, they are emitted to the parser in a stream.   Each token can also be attached to the notion of a "channel".  The parser is coded to only recognize the default channel, so if you want something ignored by the parser you can send the token to an alternative channel.  In this way the `TokenRewrite` stuff can keep all the whitespace in tact, but the parser can continue to ignore it.